### ISSUE TITLE
configuring json_key_io as StringIO breaks Fcmpush config if google reponds with 403

### ISSUE THREAD

**Issue by nikokon** [miyataka/fcmpush#19]
I've noticed when setting up fcmpush that it would break if getting a 403 from push call.

I noticed it since I have multiple projects and tried to send to a device token generated by the other project.

I got the following response:
```
Fcmpush::Forbidden: Receieved an error response 403 Forbidden: {
  "error": {
    "code": 403,
    "message": "SenderId mismatch",
    "status": "PERMISSION_DENIED",
    "details": [
      {
        "@type": "type.googleapis.com/google.firebase.fcm.v1.FcmError",
        "errorCode": "SENDER_ID_MISMATCH"
      }
    ]
  }
}
```

subsequent calls to Fcmpush.new(project_id) would throw this error: `MultiJson::ParseError: compile error`

This is because I guess it was trying to re-auth against google and thus calling .read on json_key_io. 
However reading the same StringIO multiple times would just return empty string on subsequent calls.
```
2.4.10 :005 > sio = StringIO.new("test")
 => #<StringIO:0x00007fc49d683670>
2.4.10 :006 > sio.read
 => "test"
2.4.10 :007 > sio.read
 => ""
```

There might be some other edgecases when specifying json_key_io as an instance of StringIO

Is there some other ruby-class that would provide the same interface as IO but keeping string buffer upon read?

My ugly hack as of now was just to wrap the string in a struct with an accessor for the attribute read

`config.json_key_io = Struct.new(:read).new credentials`

**Comment by miyataka** [comment]
TODO: 
- repeatable configuration

**Comment by miyataka** [comment]
@nikokon 
Fixed this. And I released as 1.3.4.
Thank you for your report.
